manual.adoc

```diff
@@ -211,5 +211,112 @@ In the example above, we've created two registries:
 
 Both registry have a `[shout]` block extension registered with a specific implementation.
 
-The first block extension is registered in the `registryA` and will transform the text to uppercase. 
+The first block extension is registered in the `registryA` and will transform the text to uppercase.
 The other one is registered in the `registryB` and will replace `.` by `{nbsp}!!!`.
+
+=== How to compile a Ruby extension to JavaScript (Node.js)
+
+Let's say you want to use the following Asciidoctor extension (written in Ruby) in your JavaScript based application.
+
+.lib/tel-inline-macro.rb
+```ruby
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal' # <1>
+
+include ::Asciidoctor
+
+# An inline macro that turns URIs with the tel: URI scheme into links.
+#
+# Usage
+#
+#   tel:1-800-555-1212[]
+#
+Extensions.register do
+  inline_macro do
+    named :tel
+    parse_content_as :text
+    process do |parent, target, attrs|
+      if (text = attrs['text']).empty?
+        text = target
+      end
+      target = %(tel:#{target})
+      (create_anchor parent, text, type: :link, target: target).render
```

Comment:
As of 1.5.6, it's no longer necessary to invoke the render method before returning. This future proofs the extension.

Reply:
Removed